src/core/agent/services/identifierService.ts

```diff
@@ -191,10 +195,22 @@ class IdentifierService extends AgentService {
     const metadata = await this.identifierStorage.getIdentifierMetadata(
       identifier
     );
-    this.validArchivedIdentifier(metadata);
-    await this.identifierStorage.updateIdentifierMetadata(identifier, {
-      isDeleted: true,
-    });
+
+    if (metadata.groupMetadata) {
+      await this.identifierStorage.deleteIdentifierMetadata(identifier);
```

Comment:
Why is there and if else statement where we delete the record here but soft delete with `isDeleted` below? We should be soft deleting or the sync with KERIA will restore it.

Reply:
@iFergal In the case of single-sig, do we still use soft delete?

Reply:
always soft delete (unless its a stale identifier)